### PR TITLE
core/query: define structs for annotated txs

### PR DESCRIPTION
**Pull request by jbowens**
Define explicit struct for annotated transactions. Annotators operate on these structs instead of the less structured `map[string]interface{}`. Defining explicit structs has the limitation that the core/query package needs to know about all possible top-level attributes. But, that will already be the case when we migrate away from relying on Postgres's jsonb type.

### CODE REVIEW

**Comment by erykwalder** [comment]
:eyes:

File: core/account/annotate.go [hunk 1]
```diff
@@ -3,61 +3,46 @@ package account
...
-			if !ok {
-				log.Error(ctx, errors.Wrap(fmt.Errorf("expected map, got %T", m)))
+	for _, tx := range txs {
+		for _, in := range tx.Inputs {
+			if len(in.ControlProgram) == 0 {
 				continue
 			}
-			if asMap["control_program"] == nil {
-				// Issuance inputs don't have control_programs
+			if controlProgramSet[string(in.ControlProgram)] {
```
Comment (jbowens) [inline core/account/annotate.go]: I think that happens down below? 
```go
	inputs[string(in.ControlProgram)] = append(inputs[string(in.ControlProgram)], in)
```
or do I misunderstand?

File: core/asset/annotate.go [hunk 2]
```diff
@@ -2,81 +2,68 @@ package asset
...
-				continue
-			}
-			assetIDStr, ok := out["asset_id"].(string)
-			if !ok {
-				log.Error(ctx, errors.Wrap(fmt.Errorf("bad asset_id type %T", out["asset_id"])))
-				continue
-			}
-			assetIDStrMap[assetIDStr] = true
+		for _, out := range tx.Outputs {
+			assetIDStrMap[hex.EncodeToString(out.AssetID)] = out.AssetID
```
Comment (jbowens) [inline core/asset/annotate.go]: Good call

File: core/asset/annotate_test.go [hunk 3]
```diff
@@ -16,95 +20,60 @@ func TestAnnotateTxs(t *testing.T) {
...
 			},
 		},
 	}
-
 	err = reg.AnnotateTxs(ctx, txs)
 	if err != nil {
 		t.Fatal(err)
 	}
 	if !reflect.DeepEqual(txs, want) {
+		spew.Dump(txs)
```
Comment (jbowens) [inline core/asset/annotate_test.go]: oops, yeah. left these in from debugging. i'll incorp into the test error

File: core/account/annotate.go (hunk 1 above)
Comment (erykwalder) [inline core/account/annotate.go]: Wouldn't the continue skip that?

Comment (jbowens) [inline core/account/annotate.go]: Oh, gotcha now. Good catch

**Comment by erykwalder** [comment]
LGTM

**Pull request merged**